circuits/transfer/src/withdraw_from_obfuscated.rs

```diff
@@ -26,42 +39,104 @@ pub struct WithdrawFromObfuscatedCircuit {
 
 impl WithdrawFromObfuscatedCircuit {
     pub fn new(
-        message_r: JubJubScalar,
-        message_ssk: &SecretSpendKey,
-        message: &Message,
+        input: &Message,
```

Comment:
Why so many by reference when they are subsequently copied in the impl?

Reply:
Also, as pointed out by @ZER0, there's some data here that seems to be flattened from other structs.

Reply:
The value opening for the circuit will not necessarily handle messages. In the WFO case, we have the same structure both for notes (output) and messages (input).

I refactored the structures organization to make it clear in the code.